### ISSUE TITLE
Godot 4.0 stable version report error log

### ISSUE THREAD

**Issue by pcloves** [Atlinx/Godot-Mono-CustomResourceRegistry#8]
When I try enable this plugin on an almost empty Godot 4.0 project, the following error log reported:

```
--- Debugging process started ---
Godot Engine v4.0.stable.mono.official.92bee43ad - https://godotengine.org
OpenGL API 3.3.0 NVIDIA 511.79 - Compatibility - Using Device: NVIDIA Corporation - NVIDIA GeForce RTX 2060 SUPER
 
--- Debugging process stopped ---
Copying Godot Offline Packages...
  modules/mono/glue/runtime_interop.cpp:1303 - System.InvalidOperationException: The type is not supported for conversion to/from Variant: 'System.Object'
     at Godot.NativeInterop.VariantUtils.GenericConversion`1.ToVariant(T& from) in /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/VariantUtils.generic.cs:line 17
     at MonoCustomResourceRegistry.Settings.AddSetting(String title, Type type, Object value, PropertyHint hint, String hintString) in D:\data\OneDrive\文档\Test\addons\MonoCustomResourceRegistry\Settings.cs:line 43
     at MonoCustomResourceRegistry.Settings.Init() in D:\data\OneDrive\文档\Test\addons\MonoCustomResourceRegistry\Settings.cs:line 29
     at MonoCustomResourceRegistry.Plugin._EnterTree() in D:\data\OneDrive\文档\Test\addons\MonoCustomResourceRegistry\Plugin.cs:line 34
     at Godot.Node.InvokeGodotClassMethod(godot_string_name& method, NativeVariantPtrArgs args, godot_variant& ret) in /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Node.cs:line 1746
     at Godot.EditorPlugin.InvokeGodotClassMethod(godot_string_name& method, NativeVariantPtrArgs args, godot_variant& ret) in /root/godot/modules/mono/glue/GodotSharp/GodotSharpEditor/Generated/GodotObjects/EditorPlugin.cs:line 1293
     at MonoCustomResourceRegistry.Plugin.InvokeGodotClassMethod(godot_string_name& method, NativeVariantPtrArgs args, godot_variant& ret) in D:\data\OneDrive\文档\Test\Godot.SourceGenerators\Godot.SourceGenerators.ScriptMethodsGenerator\MonoCustomResourceRegistry.Plugin_ScriptMethods.generated.cs:line 61
     at Godot.Bridge.CSharpInstanceBridge.Call(IntPtr godotObjectGCHandle, godot_string_name* method, godot_variant** args, Int32 argCount, godot_variant_call_error* refCallError, godot_variant* ret) in /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/CSharpInstanceBridge.cs:line 24
 
```

reproduce：
1. create a new project
2. create a Node2d scene and select as main scene
3. Project -> Tools -> C# -> Create C# project
4. unzip the plugin and enable it.
5. the error log output.

**Comment by Atlinx** [comment]
Have you tried using the `godot-4.x` branch? The main branch is still on Godot 3.x.

**Comment by pcloves** [comment]
> Have you tried using the `godot-4.x` branch? The main branch is still on Godot 3.x.

Yes, what am I using is godot-4.x, I had pulled a request, it seems to fix this issue.

**Comment by Atlinx** [comment]
Fixed by PR #9
